crates/biome_cli/src/changed.rs

```diff
@@ -0,0 +1,74 @@
+use std::process::{Command, Output};
+
+use biome_deserialize::StringSet;
+use biome_service::{configuration::FilesConfiguration, settings::to_matcher, Configuration};
+
+use crate::CliDiagnostic;
+
+pub(crate) fn store_changed_files(
+    configuration: &mut Configuration,
+    since: Option<String>,
+) -> Result<(), CliDiagnostic> {
+    let default_branch = configuration
+        .vcs
+        .as_ref()
+        .and_then(|v| v.default_branch.as_ref());
+
+    let base = match (since.as_ref(), default_branch) {
+        (Some(since), Some(_)) => since,
+        (Some(since), None) => since,
+        (None, Some(branch)) => branch,
+        (None, None) => return Err(CliDiagnostic::incompatible_end_configuration("The `--changed` flag was set, but Biome couldn't determine the base to compare against. Either set configuration.vcs.defaultBranch or use the --since argument.")),
+    };
+
+    let output = run_git_diff(base)?;
+
+    if !output.status.success() {
+        todo!("Handle error when git command fails");
```

Comment:
Not sure how this error should be handled, none of the variants on `CliDiagnostic` seemed fitting and I didn't want to just add a new arbitrary error on my own.